src/uploads/infer_image_track.py

```diff
@@ -21,40 +21,47 @@ def get_release_from_codename(codename: str) -> str:
     ].split()[1]
 
 
-parser = argparse.ArgumentParser()
-parser.add_argument(
-    "--recipe-dirname",
-    help="Path to the directory where rockcraft.yaml is",
-    required=True,
-)
-args = parser.parse_args()
-
-with open(
-    f"{args.recipe_dirname.rstrip('/')}/rockcraft.yaml", encoding="UTF-8"
-) as rockcraft_file:
-    rockcraft_yaml = yaml.load(rockcraft_file, Loader=yaml.BaseLoader)
-
-rock_base = (
-    rockcraft_yaml["base"]
-    if rockcraft_yaml["base"] != "bare"
-    else rockcraft_yaml["build-base"]
-)
-
-try:
-    base_release = float(rock_base.replace(":", "@").split("@")[-1])
-except ValueError:
-    logging.warning(
-        f"Could not infer ROCK's base release from {rock_base}. Trying with codename."
+def get_base_and_track(rockcraft_yaml) -> tuple[str, str]:
+    rock_base = (
+        rockcraft_yaml["base"]
```

Comment:
you should check for build-base first and use it if it's available, otherwise use base.
the reason is the project can have something like build-base: 22.04 and base: 24.04, but the build-base (22.04) is what you want to get.

Reply:
This change is just a refactor of the code. This logic is not touched.